src/mono/CMakeLists.txt

```diff
@@ -24,6 +24,11 @@ if (MSVC)
   # FIXME we should investigate these mismatches at some point
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/wd4245>) # W4: signed/unsigned mismatch
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/wd4389>) # W4: signed/unsigned mismatch
+
+  # these warnings come from llvm headers
+  add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/wd4244>) # W2, W3, W4: 'argument' : conversion from 'type1' to 'type2', possible loss of data, 'conversion' conversion from 'type1' to 'type2', possible loss of data
```

Comment:
Disabling 4244 will be a problem for SDL signoff (see https://github.com/dotnet/runtime/issues/66154).

cc @GrabYourPitchforks

Reply:
Interesting, I was not aware of that.

@lateralusX I see you were fixing 4244 warnings in the past in the mono sources. Could you please take a look? Should we fix the llvm headers in our dotnet/llvm-project repo?

Reply:
I think @steveisok knows the current situation around LLVM sources and warnings, I think it has been handled in some other forum in the past.

Reply:
@radekdoulik How many warnings are we talking about?

Reply:
Regardless our current strategy regarding LLVM source files, we should not disable this warning for all compiled source files.

Reply:
Yes, it is clear now that we cannot disable 4244.

I see 61 of these (C4244) warnings in the log.

```
runtime % grep C4244 warnings.txt | wc -l
      61
```

@steveisok how would we handle these?

Reply:
Are the other warnings also only from llvm headers? If so, we shouldn't disable them for all source either, but instead disable them where we include the headers (should be limited to a smaller set of source files).

Reply:
Or we can fix them together with the C4244. There are just few occurrences.
```
runtime % grep C4459 warnings.txt | wc -l
       3
runtime % grep C4127 warnings.txt | wc -l
       1

```

Reply:
I'm with @lateralusX. Disabling where we include the files is the correct place.

Reply:
@steveisok what about the C4244? How do you prefer to handle these? (in follow up PR)